packages/astro/astro-jsx.d.ts

```diff
@@ -0,0 +1,1086 @@
+/// <reference lib="dom" />
+/* eslint @typescript-eslint/no-unused-vars: off */
+/**
+ * Adapted from jsx-dom
+ * @see https://github.com/proteriax/jsx-dom/blob/be06937ba16908d87bf8aa4372a3583133e02b8a/index.d.ts
+ *
+ * which was adapted from
+ *
+ * Adapted from React’s TypeScript definition from DefinitelyTyped.
+ * @see https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react/index.d.ts
+ */
+declare namespace astroHTML.JSX {
+	/* html jsx */
+	export type Child = Node | Node[] | string | number | boolean | null | undefined | unknown;
+	export type Children = Child | Child[];
+
+	interface ElementChildrenAttribute {
+		// eslint-disable-next-line @typescript-eslint/ban-types
+		children: {};
+	}
+
+	interface IntrinsicAttributes extends AstroBuiltinProps, AstroBuiltinAttributes {
+		slot?: string;
+		children?: Children;
+	}
+
+	type AstroBuiltinProps = import('astro').AstroBuiltinProps;
+	type AstroBuiltinAttributes = import('astro').AstroBuiltinAttributes
+	type AstroDefineVarsAttribute = import('astro').AstroDefineVarsAttribute;
+	type AstroScriptAttributes = import('astro').AstroScriptAttributes & AstroDefineVarsAttribute
+	type AstroStyleAttributes = import('astro').AstroStyleAttributes & AstroDefineVarsAttribute;
+
+	// Certain Astro methods returns AstroComponent using AstroComponentFactory
+	// The language-server does not really like this because it expects Elements to
+	// all be compatible with HTMLElement so we'll add AstroComponentFactory as a valid element type
+	type AstroComponent = import('astro').AstroComponentFactory;
+	type Element = HTMLElement | AstroComponent;
```

Comment:
This change is necessary to support MarkdownInstance's `Content`. In my testing it didn't seem to create any issues to have `AstroComponent` as an alternative type for `Element`